src/core/io/src/4C_io_exodus.cpp

```diff
@@ -20,6 +21,24 @@
 
 FOUR_C_NAMESPACE_OPEN
 
+// Helper macro to always check the return value of an Exodus call.
+#define CHECK_EXODUS_CALL(call)                                                                \
+  do                                                                                           \
+  {                                                                                            \
+    int error = call;                                                                          \
+    if (error < 0)                                                                             \
+    {                                                                                          \
+      /* Use EnumTools for a direct interpretation of the error value. */                      \
+      auto error_enum = static_cast<ex_error_return_code>(error);                              \
+      FOUR_C_THROW("Exodus II failed with error code {}: {}", error_enum, #call);              \
+    }                                                                                          \
+    if (error > 0)                                                                             \
+    {                                                                                          \
+      Core::IO::cout << std::format("Exodus II returned a warning code {}: {}", error, #call); \
+    }                                                                                          \
+  } while (false)
```

Comment:
```suggestion
#define CHECK_EXODUS_CALL(call)
  do
  {
    int error = call;
    if (error < 0)
    {
      /* Use EnumTools for a direct interpretation of the error value. */
      auto error_enum = static_cast<ex_error_return_code>(error);
      FOUR_C_THROW("Exodus II failed with error code {}: {}", error_enum, #call);
    }
    if (error > 0)
    {
      Core::IO::cout << std::format("Exodus II returned a warning code {}: {}", error, #call);
    }
  } while (false)
```

can these be removed? I've never seen something like this before

Reply:
Thanks for the explanation:)